core/modules/comment/comment.install

```diff
@@ -66,6 +66,9 @@ function comment_uninstall() {
   if (db_field_exists('node', 'comment_close_override')) {
     db_drop_field('node', 'comment_close_override');
   }
+  // Reset the cached information about entity types.
+  backdrop_static_reset('entity_get_info');
+  cache()->deletePrefix('entity_info:');
```

Comment:
```suggestion
  entity_info_cache_clear()
```

Reply:
My concern regarding function entity_info_cache_clear() was that it's defined in entity.module and that function might not be available at that point.

Functions backdrop_static_reset (bootstrap.inc) and BackdropCacheInterface::deletePrefix (cache.inc) should be available.
However, @docwilmot seems to have discovered another problem (race condition?) even if the cache flush is run in comment.install. So we can't avoid to dig deeper.